### PR TITLE
youtubeuploader 18.12 (new formula)

### PR DESCRIPTION
**Pull request by sidneys**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/youtubeuploader.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    path = buildpath/"src/github.com/porjo/youtubeuploader"
+    path.install Dir["*"]
+    cd path do
+      system "dep", "ensure", "-vendor-only"
+      system "go", "build", "-o", "#{bin}/youtubeuploader"
```
Comment (DomT4) [inline Formula/youtubeuploader.rb]: You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.

**Review by DomT4 — CHANGES_REQUESTED** [review]

File: Formula/youtubeuploader.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    end
+  end
+
+  test do
+    system "#{bin}/youtubeuploader", "-v"
```
Comment (DomT4) [inline Formula/youtubeuploader.rb]: Any chance of a test that exercises a deeper level of functionality? Thanks!

Comment (sidneys) [reply Formula/youtubeuploader.rb]: @DomT4 

The recently added commit extends the build command to implement correct binary versioning (as per the developers build script) and adds the standard Homebrew test via `assert_match`, validating said version.

Cheers,
S

File: Formula/youtubeuploader.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    path.install Dir["*"]
+    cd path do
+      system "dep", "ensure", "-vendor-only"
+      system "go", "build", "-a", "-ldflags", "-s -X main.appVersion=#{version}", "-o", "#{bin}/youtubeuploader"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/youtubeuploader -v")
```
Comment (fxcoudert) [inline Formula/youtubeuploader.rb]: We need a test that exercises the some of the functionality of the app. Maybe tries to perform an upload, without adequate credentials, and fail (matching the error message and exit code)?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Pull request closed without merging**